Mathlib/Data/Sym/Sym2.lean

```diff
@@ -794,3 +796,20 @@ lemma mul_mk {M} [CommMagma M] (xy : M × M) :
     mul (.mk xy) = xy.1 * xy.2 := rfl
 
 end Sym2
+
+namespace Set
+
+open Sym2
+
+/--
+For a set `s : Set α`, `s.sym2` is the set of all unordered pairs of elements from `s`.
+-/
+def sym2 (s : Set α) : Set (Sym2 α) := fromRel (r := fun x y ↦ x ∈ s ∧ y ∈ s) (fun _ _ => .symm)
+
+@[simp] lemma mk_mem_sym2_iff {s : Set α} {x y : α} : s(x, y) ∈ s.sym2 ↔ x ∈ s ∧ y ∈ s := Iff.rfl
```

Comment:
I guess there's another version here
```lean
@[simp] lemma mk'_mem_sym2_iff {s : Set α} {xy : Prod α α} : Sym2.mk xy ∈ s.sym2 ↔ xy ∈ s.prod s := Iff.rfl
```
(with notation).

I suppose you could also add a lemma that `s.sym2 = Sym2.mk '' s.prod s`

Reply:
Making that `simp` seems to break confluence since `mem_prod` isn't simp (which means other proofs in this PR stop working), but the lemmas are helpful, added both.